okio/native/src/main/kotlin/okio/ByteString.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okio
+
+import okio.common.COMMON_EMPTY
+import okio.common.COMMON_HEX_DIGITS
+import okio.common.commonBase64
+import okio.common.commonBase64Url
+import okio.common.commonCompareTo
+import okio.common.commonDecodeBase64
+import okio.common.commonDecodeHex
+import okio.common.commonEncodeUtf8
+import okio.common.commonEndsWith
+import okio.common.commonEquals
+import okio.common.commonGetByte
+import okio.common.commonGetSize
+import okio.common.commonHashCode
+import okio.common.commonHex
+import okio.common.commonIndexOf
+import okio.common.commonInternalArray
+import okio.common.commonOf
+import okio.common.commonRangeEquals
+import okio.common.commonStartsWith
+import okio.common.commonSubstring
+import okio.common.commonToAsciiLowercase
+import okio.common.commonToAsciiUppercase
+import okio.common.commonToByteArray
+import okio.common.commonUtf8
+
+/**
+ * An immutable sequence of bytes.
+ *
+ * Byte strings compare lexicographically as a sequence of **unsigned** bytes. That is, the byte
+ * string `ff` sorts after `00`. This is counter to the sort order of the corresponding bytes,
+ * where `-1` sorts before `0`.
+ */
+actual open class ByteString
+// Trusted internal constructor doesn't clone data.
+internal actual constructor(
+    internal actual val data: ByteArray
```

Comment:
nit: 2-space indent

Reply:
Changed. `substring(...)` and `lastIndexOf(...)` use a 4 space continuation while parameters use a 2 space. Should I change these methods as well? It makes IntelliJ a little happier if the continuation indentations are consistent since this file was just a copy/paste from the JS module.

Reply:
Yeah, that was probably an oversight. Parameters should be indented by 2 spaces.